pkg/externalresource/external_resource.go

```diff
@@ -0,0 +1,65 @@
+package externalresource
+
+import (
+	b64 "encoding/base64"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/afero"
+)
+
+// ExternalResources represents the map of external resources at a manifest
+type ExternalResourceSection map[string]*ExternalResource
+
+// ExternalResource represents information on an external resource
+type ExternalResource struct {
+	Icon      string
+	Notes     Notes
+	Endpoints []ExternalEndpoint
+}
+
+// Notes represents information about the location and content of the external resource markdown
+type Notes struct {
+	Path     string
+	Markdown string // base64 encoded content of the path
+}
+
+// ExternalEndpoint represents information about an endpoint
+type ExternalEndpoint struct {
+	Name string
+	Url  string
+}
+
+// ERFilesystemManager represents ExternalResource information with the filesystem inyected
+type ERFilesystemManager struct {
+	ExternalResource ExternalResource
+	Fs               afero.Fs
+}
+
+// SetDefaults creates the necessary environment variables given an external resource
+func (er *ExternalResource) SetDefaults(externalName string) {
+	sanitizedExternalName := sanitizeForEnv(externalName)
+	for _, endpoint := range er.Endpoints {
+		sanitizedEndpointName := sanitizeForEnv(endpoint.Name)
+		endpointUrlEnv := fmt.Sprintf("OKTETO_EXTERNAL_%s_ENDPOINTS_%s_URL", sanitizedExternalName, sanitizedEndpointName)
+		os.Setenv(endpointUrlEnv, endpoint.Url)
+	}
+}
+
+func sanitizeForEnv(name string) string {
+	return strings.ToUpper(strings.ReplaceAll(name, "-", "_"))
+}
+
+// LoadMarkdownContent loads and store markdown content related to external resource
+func (ef *ERFilesystemManager) LoadMarkdownContent(manifestPath string) error {
```

Comment:
This is only used in the test. Why do we have these functions?

Reply:
yes at the moment these functions are not used at any place but both will be used by the commands. It is due to the way the epic work has been split up

Reply:
Shouldn't/can't be part of the manifest deserialization? I mean, if we are reading all the `ExternalResource` from the manifest definition, wouldn't it make sense to read the markdown there? If not, where would we be reading it?

Reply:
I need to know somehow the path of the manifest to build the path to the markdown and that inside the deserialization process I can't know it I think

Reply:
oh, ok, I see. Makes sense then to have it outside the parsing